include/bbp/sonata/output.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <highfive/H5File.hpp>
+
+#include <bbp/sonata/population.h>
+
+namespace H5 = HighFive;
+
+namespace bbp {
+namespace sonata {
+
+struct SONATA_API DataFrame {
+    using DataType = std::map<NodeID, std::vector<float>>;
+    DataType data;
+    std::vector<float> index;
+};
+
+/**
+  const SpikeReader file(filename);
+  auto pops = file.getPopulationNames();
+  for (const auto& data: file[pops[0]].get(Selection{12UL, 34UL, 58UL})) {
+      NodeID node_id;
+      double timestamp;
+      std::tie(node_id, timestamp) = data;
+      std::cout << "[" << timestamp << "] " << node_id << std::endl;
+  }
+*/
+class SONATA_API SpikeReader
+{
+  public:
+    class Population
+    {
+      public:
+        enum class Sorting : char {
+            none = 0,
+            by_id = 1,
+            by_time = 2,
+        };
+
+        using Spike = std::pair<NodeID, double>;
+        using Spikes = std::vector<Spike>;
+        Spikes get(const Selection& node_ids = Selection({}),
+                   double tstart = -1,
+                   double tstop = -1) const;
+        Sorting getSorting() const;
+
+      private:
+        Population(const std::string& filename, const std::string& populationName);
+
+        Spikes spikes;
```

Comment:
private members with _ at the end for consistency with the rest of the codebase (spikes_)